src/server/db_slice.h

```diff
@@ -330,12 +391,13 @@ class DbSlice {
   }
 
   // Check whether 'it' has not expired. Returns it if it's still valid. Otherwise, erases it
-  // from both tables and return PrimeIterator{}.
+  // from both tables and return Iterator{}.
   struct ItAndExp {
     PrimeIterator it;
```

Comment:
Shouldnt we change this as well PrimeIterator ->  Iterator and ExpireIterator -> ExpIterator ?

Reply:
The thing is, we call this also in places where we have just `PrimeIterator` in our hands, like when we search directly in the table (not through DbSlice). I thought it would be better to leave as is, as it won't have any benefit to change.
wdyt?

Reply:
Well its confusing and can lead to bugs but I think we can revisit this later

Reply:
Can you give an example for a bug it could lead to?
Because it's strongly typed, I'm not sure what you think of. I am happy to change it now instead of later though.

Reply:
If we have places in which we call ExpireIfNeeded when we call with Iterator and not PrimeIterator and than we will do another Find call. I believe we dont have such cases right now but the fact that we do such api that we enable calling ExpireIfNeeded with Iterator can lead to such bugs.

Reply:
Right, that makes sense. Done.